e2e/framework.go

```diff
@@ -422,8 +423,11 @@ func (f *fsm) IsStuck(num uint64) (uint64, bool) {
 }
 
 func (f *fsm) BuildProposal() (*pbft.Proposal, error) {
+	// make different proposal for each sequence/round
+	prop := make([]byte, 8)
+	_, _ = rand.Read(prop)
```

Comment:
You're using a weak number generator here (nitpick), but also not checking the errors it gives you

Reply:
> You're using a weak number generator here (nitpick - you can use crypto/rand)

It is fine for our use case, since we need different proposal from each round.

> but also not checking the errors it gives you

It is never returning an error, it is there as a return value because of interface. 
From the documentation: "Read generates len(p) random bytes from the default Source and writes them into p. **It always returns len(p) and a nil error**."